File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file229.f90
subroutine init_array(n, alpha, a, b)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n, n) :: b
DATA_TYPE :: alpha
integer :: n
integer :: i, j

alpha = 32412D0
do i = 1, n
do j = 1, n
a(j, i) = (DBLE(i - 1) * DBLE(j - 1)) / DBLE(n)
b(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(n)
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file229.cpp
void init_array(int n, double alpha, double a[][n], double b[][n])
{
 alpha = 32412.0;
 for (int i = 0; i < n; i++)
 for (int j = 0; j < n; j++)
 {
 a[j][i] = ((double)(i) * (double)(j)) / n;
 b[j][i] = ((double)(i) * (double)(j)) / n;
 }
}
